### PR TITLE
Post process trace

### PR DESCRIPTION
**Pull request by vb000**
Script to post process cycle-accurate HW trace to generate more human-friendly instruction by instruction execution trace.

This script takes the HW trace like this:
```
   56900  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   56920  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   56940  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   56960  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   56980  0  1: 00001290 11c000ef x01=00001294               | FP_BUBBLE                      | bt=00000124                  | 
   57000  0  1: 0000011c 00001737 x14=00001000               | FP_BUBBLE                      |                              | 
   57020  0  1: 00000120 200007b7 x15=20000000               | FP_BUBBLE                      |                              | 
   57040  0  1: 00000124 02870593 x11=00001028               | FP_BUBBLE                      |                              | 
   57060  0  1: 00000128 000016b7 x13=00001000               | FP_BUBBLE                      |                              | 
   57080  0  1: 0000012c 00b7e5b3 x11=20001028               | FP_BUBBLE                      |                              | 
   57100  0  1: 00000130 02468613 x12=00001024               | FP_BUBBLE                      |                              | 
   57120  0  1: 00000134 0005a023                            | FP_BUBBLE                      |                              | RS[20001028]=00000000
   57140  0  1: 00000138 00c7e633 x12=20001024               | FP_BUBBLE                      |                              | 
   57160  0  1: 0000013c 00062023                            | FP_BUBBLE                      |                              | RS[20001024]=00000000
   57180  0  1: 00000140 02872783 x15=00000000               | FP_BUBBLE                      |             LL[00a]=00000000 | 
   57200  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57220  0  1: 00000144 1407c063                            | FP_BUBBLE                      | bt=00000148                  | 
   57240  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57260  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57280  0  1: 00000148 0246a783 x15=00000000               | FP_BUBBLE                      |             LL[009]=00000000 | 
   57300  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57320  0  1: 0000014c 1407c463                            | FP_BUBBLE                      | bt=00000150                  | 
   57340  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57360  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57380  0  1: 00000150 02872783 x15=00000000               | FP_BUBBLE                      |             LL[00a]=00000000 | 
   57400  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57420  0  1: 00000154 1c078662                            | FP_BUBBLE                      | bt=000001cc                  | 
   57440  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57460  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57480  0  1: 000001cc 0246a783 x15=00000000               | FP_BUBBLE                      |             LL[009]=00000000 | 
   57500  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57520  0  1: 000001d0 14079c63                            | FP_BUBBLE                      | bt=000001d4                  | 
   57540  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57560  0  1: BUBBLE                                       | FP_BUBBLE                      |                              | 
   57580  0  1: 000001d4 0005a023                            | FP_BUBBLE                      |                              | RS[20001028]=00000000
   57600  0  1: 000001d8 00062023                            | FP_BUBBLE                      |                              | RS[20001024]=00000000
   57620  0  1: 000001dc 1580006f                            | FP_BUBBLE                      | bt=00000160                  | 
   57640  0  1: 00000158 02872583 x11=00000000               | FP_BUBBLE                      |             LL[00a]=00000000 | 
   57660  0  1: 0000015c 0246a683 x13=00000000               | FP_BUBBLE                      |             LL[009]=00000000 | 
   57680  0  1: 00000160 20020737 x14=20020000               | FP_BUBBLE                      |                              | 
   57700  0  1: 00000164 01259613 x12=00000000               | FP_BUBBLE                      |                              | 
   57720  0  1: 00000168 01869793 x15=00000000               | FP_BUBBLE                      |                              | 
   57740  0  1: 0000016c 00c7e7b3 x15=00000000               | FP_BUBBLE                      |                              | 
```
And, generates a concise SW debug friendly trace like below:
```
   53320  0  1: 00001268 ( fe010113 ) x02=00001fdc addi x2,x2,-32
   56860  0  1: 0000128c ( 00001a37 ) x20=00001000 lui  x20,0x1
   56980  0  1: 00001290 ( 11c000ef ) x01=00001294 jal  x1,11c <bsg_set_tile_x_y>
   57000  0  1: 0000011c ( 00001737 ) x14=00001000 lui  x14,0x1
   57020  0  1: 00000120 ( 200007b7 ) x15=20000000 lui  x15,0x20000
   57040  0  1: 00000124 ( 02870593 ) x11=00001028 addi x11,x14,40
   57060  0  1: 00000128 ( 000016b7 ) x13=00001000 lui  x13,0x1
   57080  0  1: 0000012c ( 00b7e5b3 ) x11=20001028 or   x11,x15,x11
   57100  0  1: 00000130 ( 02468613 ) x12=00001024 addi x12,x13,36
   57140  0  1: 00000138 ( 00c7e633 ) x12=20001024 or   x12,x15,x12
   57180  0  1: 00000140 ( 02872783 ) x15=00000000 lw   x15,40(x14)
   57280  0  1: 00000148 ( 0246a783 ) x15=00000000 lw   x15,36(x13)
   57380  0  1: 00000150 ( 02872783 ) x15=00000000 lw   x15,40(x14)
   57480  0  1: 000001cc ( 0246a783 ) x15=00000000 lw   x15,36(x13)
   57640  0  1: 00000158 ( 02872583 ) x11=00000000 lw   x11,40(x14)
   57660  0  1: 0000015c ( 0246a683 ) x13=00000000 lw   x13,36(x13)
   57680  0  1: 00000160 ( 20020737 ) x14=20020000 lui  x14,0x20020
   57700  0  1: 00000164 ( 01259613 ) x12=00000000 slli x12,x11,0x12
   57720  0  1: 00000168 ( 01869793 ) x15=00000000 slli x15,x13,0x18
   57740  0  1: 0000016c ( 00c7e7b3 ) x15=00000000 or   x15,x15,x12
   57760  0  1: 00000170 ( 00470613 ) x12=20020004 addi x12,x14,4
```

### CODE REVIEW

**Comment by tommydcjung** [comment]
I see that this fixes the issue, but is this the only way to go about it? I mean we have uninitialized data as 'x' for a reason.

**Comment by vb000** [comment]
Uninitiated memory contents is x for the ease of rtl verification. But software expects a valid logic value regardless of whether it being initialized or not. So x is kind of wrong from software's perspective.

File: software/py/post_process_trace.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+#
+# Input trace format:
+# <timing info>: <int pc> <int instr> <int reg write> ... | <float pc> <float instr> <float reg write> ... |
+#
+# Output format:
+# <timing info>: <pc> ( <instr> ) [<reg write>] <disassembly of instr>
+#
+# Bandhav Veluri
+# 8/2/2019
+
```
Comment (tommydcjung) [inline software/py/post_process_trace.py]: might want to make it into a python class.

**Comment by vb000** [comment]
@tommydcjung ^^^ question for you?

**Comment by tommydcjung** [comment]
nonsynthesizeable

**Pull request merged**